app/scripts/controllers/views.js

```diff
@@ -0,0 +1,263 @@
+            var expectedUrl = '/views/' + current.schema + '/' + current.name;
+            if ($location.$$url !== expectedUrl) {
+              $location.url(expectedUrl);
+            }
+          } else {
+            $scope.view = null;
+            $scope.columns = [];
+          }
+        };
+
+        ClusterEventsHandler.register('STATE_REFRESHED',
+          'ViewDetailController',
+          updateView);
+
+        $scope.$on('$destroy', function () {
+          ClusterEventsHandler.remove('STATE_REFRESHED', 'ViewDetailController');
+        });
+
+        updateView();
+        render($state.params.schema, $state.params.name);
+      }
+    };
+  })
+  .directive('viewList', function () {
+    return {
+      restrict: 'E',
+      replace: true,
+      scope: {},
+      templateUrl: 'static/views/viewlist.html',
+      controllerAs: 'ViewListController',
+      controller: function ($scope, $state, TabNavigationInfo, ClusterEventsHandler, ClusterState) {
+        $scope.collapsed = true;
+
+        $scope.viewFilter = (item) => {
+          return (!$scope.search || (item.fqn.toLowerCase().indexOf($scope.search) != -1));
+        };
+
+        $scope.isActive = (schema, name) => {
+          return name === $state.params.name && schema === $state.params.schema;
+        };
+
+        var toggleClasses = (i) => {
+          $('#nav-tabs-' + i)
+            .collapse('toggle');
+          $('#nav-tabs-header-' + i + ' i.fa')
+            .toggleClass('fa-caret-down')
+            .toggleClass('fa-caret-right');
+        };
+
+        $scope.toggleElements = (i) => {
+          toggleClasses(i);
+          TabNavigationInfo.toggleIndex(i);
+        };
+
+        $scope.isCollapsed = (i) => {
+          return TabNavigationInfo.collapsed[i];
+        };
+
+        $scope.collapseAll = () => {
+          var idx;
+          for (idx in $scope.views) {
+            toggleClasses(idx);
+            TabNavigationInfo.collapseIndex(idx);
+          }
+          $scope.collapsed = true;
+        };
+
+        $scope.unCollapseAll = () => {
+          var idx;
+          for (idx in $scope.views) {
+            toggleClasses(idx);
+            TabNavigationInfo.unCollapseIndex(idx);
+          }
+          $scope.collapsed = false;
+        };
+
+        var filterBySchemaName = (schema) => (o) => o.schema === schema;
+
+        var updateViewList = () => {
+          var views = ClusterState.data.views;
+          var hasViews = views.length > 0;
+          $scope.renderSidebar = hasViews;
+          $scope.views = [];
+          if (hasViews) {
+            // use name and schema from first item
+            if (!$scope.name || !$scope.schema) {
+              $scope.schema = views[0].schema;
+              $scope.name = views[0].name;
+              $state.go('views.view', {
+                schema: $scope.schema,
+                name: $scope.name
+              });
+            }
+
+            var idx,
+              name;
```

Comment:
Any reason for the weird indentation here?

Reply:
that is the correct indentation (2 spaces) when having multiline variable declaration.

Reply:
Ah, cool, didnt know that.